### PR TITLE
Use Raxa X version as a default for nginx and event-service

### PR DESCRIPTION
**Pull request by tczekajlo**
Use Rasa X version (`rasax.tag` value) as a default for nginx and event-service. The PR resolves issue #15

### CODE REVIEW

File: charts/rasa-x/templates/event-service-deployment.yaml [hunk 1]
```diff
@@ -29,7 +29,11 @@ spec:
       {{- end }}
       containers:
       - name: {{ .Chart.Name }}
+        {{- if not .Values.eventService.tag }}
+        image: "{{ .Values.eventService.name }}:{{ .Values.rasax.tag | default .Chart.AppVersion }}"
```
Comment (wochinge) [inline charts/rasa-x/templates/event-service-deployment.yaml:33]: how about using a helper template for `{{ .Values.rasax.tag | default .Chart.AppVersion }}` to not repeat ourselves?
Doing this we could also unify lines 33 and 35, e.g. something along the lines:

```
        image: "{{ .Values.eventService.name }}:{{ .Values.eventService.tag | default <template function> }}"

```

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @dakshvar22 will take a look at it as soon as possible ✨

File: charts/rasa-x/templates/event-service-deployment.yaml (hunk 1 above)
Comment (tczekajlo) [reply charts/rasa-x/templates/event-service-deployment.yaml]: Sounds good! I see one place more where we can replace the `rasax.tag` variable and use a template function instead.

**Comment by wochinge** [comment]
@erohmensing Can you please add @tczekajlo to the list of Rasa team members?

File: charts/rasa-x/templates/_helpers.tpl [hunk 2]
```diff
@@ -192,3 +192,10 @@ Return the checksum of Rasa X values.
 {{- define "rasa-x.checksum" -}}
 {{ toYaml .Values.rasax | sha256sum }}
 {{- end -}}
+
+{{/*
+Return the AppVersion value as a default if the rasax.tag variable is not defined.
+*/}}
+{{- define "rasa-x.version" -}}
+{{ .Values.rasax.tag | default .Chart.AppVersion }}
+{{- end -}}
```
Comment (wochinge) [inline charts/rasa-x/templates/_helpers.tpl:201]:
```suggestion
{{- end -}}

```

**Comment by erohmensing** [comment]
@wochinge I did after i deleted it 😅

**Comment by kgorskowski** [comment]
Hi folks. I just wanted to add that this kinda breaks it for us or at least makes it more complicated. We build a modified container of rasa-x with gitlab set the rasa-x tag for helm to the commit sha. So now we have to set the nginx tag manually or it would break our helm deployment. Maybe we are using it wrong. But building our own rasa-x image was the only solution I found to add custom channels/scripts to the image.

**Comment by wochinge** [comment]
> But building our own rasa-x image was the only solution I found to add custom channels/scripts to the image.

Did you checkout https://rasa.com/docs/rasa-x/installation-and-setup/openshift-kubernetes/#configuring-rasa-open-source-channels ?

**Comment by kgorskowski** [comment]
> > But building our own rasa-x image was the only solution I found to add custom channels/scripts to the image.
> 
> Did you checkout https://rasa.com/docs/rasa-x/installation-and-setup/openshift-kubernetes/#configuring-rasa-open-source-channels ?

Little background, I myself have slim to zero knowledge how rasa works internally :) I support a team of students who are building on top of rasa-x with ci/cd and deployment stuff. The students are writing a custom message "connector" or channel. As of My understanding of https://rasa.com/docs/rasa/user-guide/messaging-and-voice-channels/ is that the custom script ( ( `custom_channel_rasa.py` ) has to be present alongside credentials.yml etc in the rasa-x container?  With the helm chart I found no practicable way to make that available other than build our own image on top of rasa-x that contains the python script.

**Comment by wochinge** [comment]
@kgorskowski Thanks for the explanation and great work! When you do the configuration as described in [here](https://rasa.com/docs/rasa-x/installation-and-setup/openshift-kubernetes/#configuring-rasa-open-source-channels), Helm will automatically mount these settings as `credentials.yml` in your Rasa deployment. So there is really no need to extend the existing Docker images.

**Comment by kgorskowski** [comment]
Hi @wochinge thanks again for you feedback.
I asked the question also in the community forum and found a "kinda" solution
https://forum.rasa.com/t/basic-question-regarding-custom-connectors-channels-and-kubernetes-deployment/29062/4
As the configuration happens in the credentials.yml (via helm values.yaml) the connector/script itself has to be present within rasa. My misunderstanding was, that I thought it has to be in the rasa-x container, but it must be present in the rasa containers (for worker/production pods). 
We have it working now, but only with building our custom image on top of the rasa image. In regards to helm deployment this could probably be streamlined with extraContainers or initContainers that are able to write a configmap to the "/app" mount or something. But as I said, for now (and regarding that this is a students project) we got it running.

**Comment by wochinge** [comment]
> We have it working now, but only with building our custom image on top of the rasa image. 

To be honest, I think this is also a cleaner approach than using side containers, e.g. you might need to install extra dependencies and so on. You could set up a CI job which builds and tests your image whenever there is a new rasa release.

**Pull request merged**